File: src/fe/src/types/ticket.ts
export type TicketType = 'Broken' | 'Empty'
export type TicketStatus = 'Pending' | 'Confirmed' | 'Done'

export interface Ticket {
  id: string
  type: TicketType
  status: TicketStatus
  productId: string | null
  freeTextDesc: string | null
  imageUrl: string | null
  deadline: string
  requesterId: string
  confirmedById: string | null
  createdAt: string
  isOverdue: boolean
  product: {
    id: string
    name: string
    categoryId: string
    categoryName: string | null
  } | null
  requester: { id: string; name: string } | null
}

export interface DeadlineHistoryEntry {
  id: string
  oldDeadline: string
  newDeadline: string
  reason: string
  changedById: string
  changedBy: { id: string; name: string } | null
  changedAt: string
}

export interface TicketDetail extends Ticket {
  confirmedBy: { id: string; name: string } | null
  deadlineHistory: DeadlineHistoryEntry[]
}

File: src/be/AllianceSupplier.Domain/Entities/Ticket.cs
using AllianceSupplier.Domain.Enums;

namespace AllianceSupplier.Domain.Entities;

public class Ticket
{
    public Guid Id { get; set; }
    public TicketType Type { get; set; }
    public TicketStatus Status { get; set; } = TicketStatus.Pending;
    public Guid? ProductId { get; set; }
    public string? FreeTextDesc { get; set; }
    public string? ImageUrl { get; set; }
    public DateTime Deadline { get; set; }
    public Guid RequesterId { get; set; }
    public Guid? ConfirmedById { get; set; }
    public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
    public DateTime UpdatedAt { get; set; } = DateTime.UtcNow;

    public Product? Product { get; set; }
    public User Requester { get; set; } = null!;
    public User? ConfirmedBy { get; set; }
    public ICollection<DeadlineHistory> DeadlineHistories { get; set; } = [];
}
